jedi/inference/gradual/annotation.py

```diff
@@ -333,25 +333,47 @@ def iter(iterable: Iterable[_T]) -> Iterator[_T]: ...
     unpacks the `Iterable`.
     """
     type_var_dict = {}
+    annotation_name = annotation_value.py__name__()
+
     if isinstance(annotation_value, TypeVar):
         if not is_class_value:
-            return {annotation_value.py__name__(): value_set.py__class__()}
-        return {annotation_value.py__name__(): value_set}
+            return {annotation_name: value_set.py__class__()}
+        return {annotation_name: value_set}
     elif isinstance(annotation_value, TypingClassValueWithIndex):
-        name = annotation_value.py__name__()
-        if name == 'Type':
+        if annotation_name == 'Type':
             given = annotation_value.get_generics()
             if given:
-                for nested_annotation_value in given[0]:
-                    _merge_type_var_dicts(
-                        type_var_dict,
-                        _infer_type_vars(
-                            nested_annotation_value,
-                            value_set,
-                            is_class_value=True,
+                if is_class_value:
```

Comment:
Just wondering, why is needed? What's the example that makes this necessary?

Reply:
I don't recall exactly, though according to https://github.com/davidhalter/jedi/commit/c03ae0315e84585c55927fe470477bd7faa9869a, this is to make nested `Type[T]` cases work. I don't seem to have added a test in that commit, so likely there already was one somewhere in the tests I added (or elsewhere).

I'll check this and add some comments around what the branches do as part of tidying this up (hopefully later this week).

Reply:
I've added a general description of `is_class_value` to 54e29eede19947c27f63cfa7929e23d47327b135, does that help?